yaml.go

```diff
@@ -94,3 +106,35 @@ func buffer(file *os.File) []byte {
 	}
 	return make([]byte, size)
 }
+
+func isURL(pathname string) bool {
+    _, err := url.ParseRequestURI(pathname)
+    if err != nil {
+        return false
+    } else {
+        return true
+	}
+}
+
+func downloadFile(pathname string) (string, error) {
+	_, filename := path.Split(pathname)
+	tmpFile, err := ioutil.TempFile(os.TempDir(), filename)
+    if err != nil {
+		log.Error(err, "Cannot create temporary file")
+		return "", err
+	}
+	defer tmpFile.Close()
+
+    resp, err := http.Get(pathname)
+    if err != nil {
+        return "", err
+    }
+    defer resp.Body.Close()
+
+    _, err = io.Copy(tmpFile, resp.Body)
+    if err != nil {
+        return "", err
+	}
+
+	return tmpFile.Name(), err
```

Comment:
You might want to do `return tmpFile.Name(), nil` to make clear there is no error in here.